distribution/pull_v2.go

```diff
@@ -68,15 +68,15 @@ type puller struct {
 	metadataService metadata.V2MetadataService
 	endpoint        registry.APIEndpoint
 	config          *ImagePullConfig
-	repoInfo        *registry.RepositoryInfo
+	repoName        reference.Named
 	repo            distribution.Repository
 	manifestStore   *manifestStore
 }
 
 func (p *puller) pull(ctx context.Context, ref reference.Named) (err error) {
 	// TODO(tiborvass): was ReceiveTimeout
 	// TODO(thaJeztah): do we need p.repoInfo at all, as it would probably be same as ref?
-	p.repo, err = newRepository(ctx, p.repoInfo.Name, p.endpoint, p.config.MetaHeaders, p.config.AuthConfig, "pull")
+	p.repo, err = newRepository(ctx, p.repoName, p.endpoint, p.config.MetaHeaders, p.config.AuthConfig, "pull")
```

Comment:
Looks like I forgot to update the comment here; let me do a quick rebase and fix that